nistats/contrasts.py

```diff
@@ -192,8 +192,8 @@ def stat(self, baseline=0.0):
                 np.maximum(self.variance, self.tiny))
         else:
             raise ValueError('Unknown statistic type')
-        self.stat_ = stat
-        return stat.ravel()
+        self.stat_ = stat.ravel()
+        return self.stat_
```

Comment:
Note that in line 213, `p_value` sets `self.stat_` to the output of `self.stat()`, where here we were outputing an `ravel`'d version of `self.stat_`. This seems to be the heart of the bug.

Reply:
Indeed ! 
OK, I can try and fix that one.

Reply:
I think this is the fix...